### PR TITLE
Update/envoy1.32

### PR DESCRIPTION
**Pull request by Stevenjin8**
envoy security release

### CODE REVIEW

File: charts/osm-arc/values.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ osm:
     deployPrometheus: false
     deployJaeger: false
     webhookConfigNamePrefix: arc-osm-webhook
-    sidecarImage: mcr.microsoft.com/oss/envoyproxy/envoy:v1.28.7
+    sidecarImage: mcr.microsoft.com/oss/envoyproxy/envoy-distroless:v1.32.2-hotfix.20241216
```
Comment (keithmattix) [inline charts/osm-arc/values.yaml]: Not sure we should move to distroless here since previous versions have had a shell. WDYT?

Comment (Stevenjin8) [inline charts/osm-arc/values.yaml]:
```suggestion
    sidecarImage: mcr.microsoft.com/oss/envoyproxy/envoy:v1.32.2-hotfix.20241216
```

**Comment by keithmattix** [comment]
Can you trigger the tests for arc?

**Comment by Stevenjin8** [comment]
@keithmattix I don't think I can. I tried to do it via ADO, but I'm getting a permissions error

**Comment by keithmattix** [comment]
azp run upgrade-e2e-arc

**Review by keithmattix — APPROVED** [review]

**Pull request merged**